src/main/java/uk/gov/hmcts/reform/sscs/services/ccd/UpdateCcdRepresentative.java

```diff
@@ -23,14 +26,44 @@ static boolean updateCcdRepresentative(SscsCaseData gapsCaseData, SscsCaseData e
                 || !existingRepresentative.getName().equals(rep.getName())
                 || !existingRepresentative.getContact().equals(rep.getContact())
             ) {
-                existingCcdCaseData.getAppeal().setRep(rep);
+                if (rep.getContact() != null) {
+                    if (rep.getContact().getMobile() != null) {
+                        rep.getContact()
+                            .setMobile(checkValidUkMobileNumberElseReturnExisting(
+                                rep, existingRepresentative, gapsCaseData.getCaseReference()));
+                    }
+                    existingCcdCaseData.getAppeal().setRep(rep);
```

Comment:
Here the representative details will be set which contains name, contact and address
so if condition check for name and contact will be not work as expected